### ISSUE TITLE
HiveObject does not have key when loaded lazy

### ISSUE THREAD

**Issue by PauloMelo** [hivedb/hive#112]
**Steps to Reproduce**
When opening a box in lazy mode the key of the HiveObject is not set, as it is when the object is opened normally.

In the code Sample bellow it shows:
0 1
null 2

**Code sample**
```
import 'dart:io';

import 'package:hive/hive.dart';

@HiveType()
class Test extends HiveObject {
  @HiveField(0)
  final int id;

  Test (this.id);
}

class TestAdapter extends TypeAdapter<Test> {
  @override
  Test read(BinaryReader reader) {
    var numOfFields = reader.readByte();
    var fields = <int, dynamic>{
      for (var i = 0; i < numOfFields; i++) reader.readByte(): reader.read(),
    };
    return Test(
      fields[0] as int,
    );
  }

  @override
  void write(BinaryWriter writer, Test obj) {
    writer
      ..writeByte(1)
      ..writeByte(0)
      ..write(obj.id);
  }
}

void main() async {
    Hive.init(Directory.current.path);
    Hive.registerAdapter(TestAdapter(), 0);

    var memBox = await Hive.openBox('normalBox');
    await memBox.add(Test(1));

    var lazyBox = await Hive.openBox('lazyBox', lazy: true) as LazyBox;
    await lazyBox.add(Test(2));

    var obj1 = memBox.getAt(0);
    print("${obj1.key} ${obj1.id}");

    var obj2 = await lazyBox.getAt(0);
    print("${obj2.key} ${obj2.id}");
}
```

**Version**
 - Platform: iOS, Android, Mac
 - Flutter version: 1.10.15
 - Hive version: 1.1.1

**Comment by leisim** [comment]
Thanks for reporting!

**Comment by leisim** [comment]
Fix will be published with `v1.2.0`
